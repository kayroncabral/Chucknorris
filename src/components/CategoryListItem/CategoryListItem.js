import React from 'react'
import PropTypes from 'prop-types'
import styles from './CategoryListItem.scss'

const CategoryListItem = ({ title, onItemClick }) => {
  const onClick = e => {
    onItemClick(e)
  }

  return (
    <div className={styles.CategoryListItem} onClick={onClick}>
      <p>{title}</p>
    </div>
  )
}

CategoryListItem.propTypes = {
  title: PropTypes.string,
  onItemClick: PropTypes.func
}

CategoryListItem.defaultProps = {
  title: '',
  onItemClick: () => {}
}

export default CategoryListItem
