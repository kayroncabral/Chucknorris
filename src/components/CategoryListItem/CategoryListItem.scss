@import '../../styles/styles.scss';

.CategoryListItem {
  padding: $l-size;
  display: flex;
  align-items: center;
  color: $quick-silver;
  background: white;
  transition: background 1.2s;
  text-transform: capitalize;
  border-bottom: 1px solid $gainsboro;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.14);

  > p {
    margin: 0;
    font-size: $font-size-xlarge;
    font-style: 200;
  }

  &:hover {
    cursor: pointer;
  }

  &:nth-child(even) {
    transform: perspective(100px) rotateX(3deg) translateY(2px) scale(1.001);
    background: linear-gradient(to bottom,  #ffffff 0%,#EFEFEF 100%);
  }

  &:nth-child(odd) {
    transform: perspective(100px) rotateX(-3deg) translateY(3px);
    background: linear-gradient(to top,  #ffffff 0%,#EFEFEF 100%);
  }
}
