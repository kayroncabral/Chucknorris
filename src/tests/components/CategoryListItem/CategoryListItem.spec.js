import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import CategoryListItem from '../../../components/CategoryListItem'

test('should render Header correctly', () => {
  const renderer = new ReactShallowRenderer()

  renderer.render(<CategoryListItem/>)
  expect(renderer.getRenderOutput()).toMatchSnapshot()
})
