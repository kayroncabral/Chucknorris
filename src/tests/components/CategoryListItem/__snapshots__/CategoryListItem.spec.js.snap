// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  className={undefined}
  onClick={[Function]}
>
  <p>
    
  </p>
</div>
`;
